tests/src/services/products.service.test.ts

```diff
@@ -228,4 +228,130 @@ describe('Products Service Tests', () => {
       });
     });
   });
+
+  describe('Custom user features overrides', () => {
+    test('When user has custom feature overrides, then they are applied to the base tier', async () => {
```

Comment:
_merged_ could be a better word to describe the mix of custom features + tier features (now you are using _applied_)

Reply:
Indeed.